q2_feature_classifier/_cutter.py

```diff
@@ -178,7 +200,9 @@ def extract_reads(sequences: DNAIterator,  f_primer: str, r_primer: str,
                 'r_primer': Str,
                 'identity': Float,
                 'min_length': Int % Range(0, None),
-                'max_length': Int % Range(0, None)},
+                'max_length': Int % Range(0, None),
+                'n_jobs': Int % Range(1, None),
+                'batch_size': Int % Range(0, None)},
```

Comment:
How about this, instead:

```suggestion
               Int % Range(1, None) | Str % Choices(['auto']),
```


cc @ebolyen, does this look right to you? I can't seem to get this to work locally (`AttributeError: <class 'qiime2.core.type.grammar.UnionExp'> object has no attribute 'predicate'`)

Reply:
Just to follow up, this only raises the `AttributeError` in a plugin registration, it is fine if I construct manually and interrogate:

```python
In [5]: x = Int % Range(1, None) | Str % Choices(['auto'])

In [6]: type(x)
Out[6]: qiime2.core.type.grammar.UnionExp

In [7]: x
Out[7]: Int % Range(1, None) | Str % Choices('auto')

In [8]: x.members
Out[8]: (Int % Range(1, None), Str % Choices('auto'))

In [9]: 'auto' in x
Out[9]: True

In [10]: 'fauto' in x
Out[10]: False

In [11]: 2 in x
Out[11]: True
```

Reply:
Oh dear... the plugin API might be touching things it ideally shouldn't...

Reply:
You think so? I am still willing to chalk it up to dev error. I could've _sworn_ we tested this case out when working on types v4....

Reply:
Issue: https://github.com/qiime2/q2cli/issues/221

Reply:
@thermokarst How exactly is this feature meant to work? What type of object is `Choices` supposed to be?